### PR TITLE
New components & updates

### PR DESCRIPTION
**Pull request by Lvlad27**
- Update Current Weather Forecast component with weather details info
- Create Daily Forecast component
- Create Search Bar component with autosuggestions from API
- Create Hourly Forecast component
- Folder and file restructure & add Footer Component
- Current Weather Forecast, Daily Forecast and Hourly Forecast (Home Page) now gets updated based on search input

image
image

### CODE REVIEW

File: src/Components/CurrWeatherCard/CurrWeatherCard.jsx [hunk 1]
```diff
@@ -13,13 +13,19 @@ export default function CurrWeatherCard({ locationData, currWeather }) {
...
+    const selectedDate = new Date(currWeather.current.time);
+
     let hour = selectedDate.toLocaleString('en-GB', {
       hour: '2-digit',
       minute: '2-digit',
       hour12: false
     });
-
-    return `${date} ${year} ${hour}`;
+    return `${hour}`;
```
Comment (JuliaShavyrova) [inline src/Components/CurrWeatherCard/CurrWeatherCard.jsx]: `toLocaleString` returns string as default, no need to wrap it in string again

File: src/Components/CurrWeatherCard/CurrWeatherCard.jsx [hunk 2]
```diff
@@ -13,13 +13,19 @@ export default function CurrWeatherCard({ locationData, currWeather }) {
     let year = selectedDate.toLocaleString('en-GB', {
       year: 'numeric'
     });
+
+    return `${date} ${year}`;
+  }
+
+  function formatTime() {
```
Comment (JuliaShavyrova) [inline src/Components/CurrWeatherCard/CurrWeatherCard.jsx]: formatDate and formatTime functions can be moved outside of the component body

File: src/Components/CurrWeatherCard/CurrWeatherCard.jsx [hunk 3]
```diff
@@ -37,8 +43,42 @@ export default function CurrWeatherCard({ locationData, currWeather }) {
         <img width="120" src={getWeatherSymbol(currWeather.current.symbol)} />
```
Comment (JuliaShavyrova) [inline src/Components/CurrWeatherCard/CurrWeatherCard.jsx]: i noticed you use everywhere `currWeather.current`, why not just pass `currWeather.current` as prop and make code a bit cleaner?

File: src/Components/DailyWeather/DailyWeatherCard.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+export default function DailyWeatherCard({ dailyWeather }) {
+  const formatDate = (dateStr) => {
+    let selectedDate = new Date(dateStr);
+
+    let date = selectedDate
+      .toLocaleString('en-GB', {
+        day: 'numeric',
+        weekday: 'short',
+        month: 'numeric'
+      })
```
Comment (JuliaShavyrova) [inline src/Components/DailyWeather/DailyWeatherCard.jsx]: you can create shared utility function to get formatted date and use it both in CurrWeatherCard and WeatherCard components

File: src/Components/CurrWeatherCard/CurrWeatherCard.jsx (hunk 1 above)
Comment (Lvlad27) [reply src/Components/CurrWeatherCard/CurrWeatherCard.jsx]: done

File: src/Components/CurrWeatherCard/CurrWeatherCard.jsx (hunk 2 above)
Comment (Lvlad27) [reply src/Components/CurrWeatherCard/CurrWeatherCard.jsx]: done

File: src/Components/CurrWeatherCard/CurrWeatherCard.jsx (hunk 3 above)
Comment (Lvlad27) [reply src/Components/CurrWeatherCard/CurrWeatherCard.jsx]: That's right, done.

File: src/Components/DailyWeather/DailyWeatherCard.jsx (hunk 4 above)
Comment (Lvlad27) [reply src/Components/DailyWeather/DailyWeatherCard.jsx]: done

File: src/Components/FeedbackPage/FeedbackPage.module.scss [hunk 5]
```diff
@@ -0,0 +1,9 @@
+@import '../../Styles/variables.scss';
+
+.container {
+  display: flex;
+  flex-direction: column;
+  width: 80%;
+  margin: auto;
+  gap: 5rem;
```
Comment (JuliaShavyrova) [inline src/Components/FeedbackPage/FeedbackPage.module.scss:8]: you can create global classname or shared component for containers styling

**Review by JuliaShavyrova — APPROVED** [review]

**Pull request merged**